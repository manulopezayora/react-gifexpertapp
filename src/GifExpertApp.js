import React, { useState } from 'react'

// Components
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories)

  return (
    <>
      <h1 className="title">GifExpertApp</h1>
      <AddCategory setCategories={setCategories} />
      <hr />

      {
        <ol>
          {categories.map((category) => (
            <GifGrid key={category} category={category} />
          ))}
        </ol>
      }
    </>
  )
}

export default GifExpertApp
