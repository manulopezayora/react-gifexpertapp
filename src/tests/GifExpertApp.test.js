import React from 'react'
import '@testing-library/jest-dom'

import { shallow } from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('Pruebas para el componente <GifExpertApp />', () => {
  test('debería mostrar el componente correctamente', () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot()
  })

  test('debe de mostrar una lista de categorías', () => {
    const categories = ['Panda', 'Dog']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
})
