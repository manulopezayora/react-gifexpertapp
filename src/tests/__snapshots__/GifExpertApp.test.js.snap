// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente <GifExpertApp /> debe de mostrar una lista de categorías 1`] = `
<Fragment>
  <h1
    className="title"
  >
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Panda"
      key="Panda"
    />
    <GifGrid
      category="Dog"
      key="Dog"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas para el componente <GifExpertApp /> debería mostrar el componente correctamente 1`] = `
<Fragment>
  <h1
    className="title"
  >
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
